llvm/lib/Object/COFFImportFile.cpp

```diff
@@ -691,10 +692,15 @@ Error writeImportLibrary(StringRef ImportName, StringRef Path,
       }
 
       if (!E.ImportName.empty() && Name != E.ImportName) {
+        StringRef Prefix = "";
+        if (Machine == IMAGE_FILE_MACHINE_I386 && AddUnderscores)
+          Prefix = "_";
```

Comment:
I think this is not exactly the same as the code removed from `COFFModuleDefinition`. The original code also checks if the symbol is not decorated before adding the prefix.

Reply:
That's true, but I'm not entirely sure if this is a case we need to care about... For mingw mode, the existing `isDecorated` only triggers for fastcall/vectorcall/MSVC C++ symbols.

The whole mechanism for how the aliases find the symbol to point to is reworked in https://github.com/llvm/llvm-project/pull/98229 anyway.

I guess I can adjust the commit message to point out that this technically isn't fully NFC, but it's practically NFC for the cases we care about here.

Reply:
Fair enough, it indeed doesn't seem too important. Looks good otherwise, thanks.

Reply:
FWIW, I traced the addition of the `isDecorated` call to 284ab80f8d055, and it doesn't necessarily seem motivated there. On current git main, we can also remove the call entirely, without affecting any testcase in either `check-llvm` or `check-lld`.

So it seems like the whole call may have been misguided to begin with.